pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -202,13 +213,20 @@ func (a *index) constructService(svc *v1.Service, w *Waypoint) *workloadapi.Serv
 		Ports:         ports,
 		Waypoint:      waypointAddress,
 		LoadBalancing: lb,
+		IpFamilies:    ipPolicy,
 	}
 }
 
 func getVIPs(svc *v1.Service) []string {
 	res := []string{}
-	if svc.Spec.ClusterIP != "" && svc.Spec.ClusterIP != v1.ClusterIPNone {
-		res = append(res, svc.Spec.ClusterIP)
+	cips := svc.Spec.ClusterIPs
+	if len(cips) == 0 {
+		cips = []string{svc.Spec.ClusterIP}
+	}
+	for _, cip := range cips {
+		if cip != "" && cip != v1.ClusterIPNone {
```

Comment:
add ut for this function

Reply:
https://github.com/istio/istio/pull/51665 adds some basic ones, will make it easier to add new ones. if we can merge that it will help and Ill update this PR with a test